server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/ImapStepdefs.java

```diff
@@ -104,10 +104,12 @@ public void openImapConnectionAndSelectMailbox(String mailbox) throws Throwable
         String login = userStepdefs.getConnectedUser();
         String password = userStepdefs.getUserPassword(login);
 
-        imapConnections.put(mailbox, new IMAPMessageReader()
-            .connect(LOCALHOST, IMAP_PORT)
-            .login(login, password)
-            .select(mailbox));
+        try (IMAPMessageReader imapMessageReader = new IMAPMessageReader()) {
```

Comment:
Ok Ahhhhhhhhh I got why you broke the build @aduprat !!!

You are closing connections that are going to be used later on.

So the tests are working way less better :trollface: 

What is the rule not respected here? Please revert that part(s) of the diff, and find a way to avoid your warning in eclipse.